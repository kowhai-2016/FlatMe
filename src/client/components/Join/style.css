/* Join Page
–––––––––––––––––––––––––––––––––––––––––––––––––– */

.Join .input-group {
  width: 340px;
  border-radius: 3px;
}

.Join .bg-join {
  background-image: url('/images/table.jpg');
  background-size:100%;
  min-height: 800px;
  background-repeat: no-repeat;
}

.Join .containerJoin {
  background-color: #fff;
}

.Join .globalJoinForm  {
  max-width: 600px;
  margin-left: auto;
  margin-right: auto;
  margin: 5%;
  text-align: right;
}

@media screen and (max-width: 767px) and (min-width: 200px) {
  .Join .globalJoinForm {
    text-align: left;
  }
}

@media screen and (max-width: 767px) and (min-width: 200px) {
  .Join .titleJoinForm label{
    padding-top: 0px;
  }
}

@media screen and (max-width: 767px) and (min-width: 200px) {
  .Join .errors {
    font-size: 0.8em;
  }
}

.Join .input-group-lg {
  margin: 2%;
  height: 30px;
}

.Join .titleJoinForm label{
  text-align: right;
  padding-top: 4%;
  padding-left: 0px;
}

.Join .submitJoinBtn {
  width: 100px;
  height: 40px;
  margin-top: 10px;
  margin-bottom: 20px;
}

.Join .joinTitle h1 {
  text-align: center;
  margin-bottom: 35px;
}

.Join .containerJoin {
  border: 1px solid grey;
  margin: 10px;
  max-width: 650px;
  margin-left: auto;
  margin-right: auto;
  margin-top: 30px;
  border-radius: 25px;
  -webkit-box-shadow: 0px 3px 20px 0px rgba(0,0,0,1);
  -moz-box-shadow: 0px 3px 20px 0px rgba(0,0,0,1);
  box-shadow: 0px 3px 20px 0px rgba(0,0,0,1);
}

.Join .errors {
  color: red;
  /*height: 20px;
  text-align: right;*/
}
