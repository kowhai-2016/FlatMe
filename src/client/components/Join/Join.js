import React, { PropTypes } from 'react'
import { Control, Errors, Form } from 'react-redux-form'
import { Col } from 'react-bootstrap'

import './style.css'

const isNumber = val => !isNaN(Number(val))
const maxLength = length => val => val && val.length <= length
const passwordMatch = vals => vals.confirmPassword === vals.password
const required = val => val && val.length

const Join = React.createClass({
  handleSubmit (values) {
    this.props.signUp(values)
    .then(user => {
      if (user) {
        document.getElementById('successful-signup-confirm').innerHTML = 'Sign up successful!'
      }
    })
  },
  render () {
    return (
      <div className='Join container-fluid'>
        <div className='row'>
          <div className='col-md-12 bg-join'>
            <div className='container containerJoin'>
              <Form model='forms.join'
                onSubmit={this.handleSubmit}
                validators={{
                  '': {
                    passwordMatch
                  }
                }}
                >
                <div className='joinTitle'>
                  <Col sm={12} >
                    <h1>Join us</h1>
                  </Col>
                </div>
                <div className='globalJoinForm'>
                  <div className='titleJoinForm'>
                    <Col sm={4} >
                      <label>First name:</label>
                    </Col>
                    <Col sm={6}>
                      <Control.text
                        className='input-group input-group-lg'
                        placeholder='Enter Your Name'
                        model='.firstName'
                        validators={{
                          required,
                          maxLength: maxLength(15)
                        }}
                      />
                      <Errors
                        className='errors'
                        model='forms.join.firstName'
                        show='touched'
                        messages={{
                          required: 'Required',
                          maxLength: ' Must be 15 characters or less'
                        }}
                      />
                    </Col>
                  </div>

                  <div className='titleJoinForm'>
                    <Col sm={4} >
                      <label>Last Name:</label>
                    </Col>
                    <Col sm={6}>
                      <Control.text
                        placeholder='Enter Your Family Name'
                        className='input-group input-group-lg'
                        model='.lastName'
                        validators={{
                          required,
                          maxLength: maxLength(15)
                        }}
                      />
                      <Errors
                        className='errors'
                        model='forms.join.lastName'
                        show='touched'
                        messages={{
                          required: 'Required',
                          maxLength: ' Must be 15 characters or less'
                        }}
                      />
                    </Col>
                  </div>

                  <div className='titleJoinForm'>
                    <Col sm={4} >
                      <label>Email:</label>
                    </Col>
                    <Col sm={6}>
                      <Control.text
                        placeholder='Enter Your Email'
                        className='input-group input-group-lg'
                        model='.email'
                        validators={{
                          required,
                          validEmail: (val) => /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(val)
                        }}
                      />
                      <Errors
                        className='errors'
                        model='forms.join.email'
                        show='touched'
                        messages={{
                          required: 'Required',
                          validEmail: ' Invalid email address'
                        }}
                      />
                    </Col>
                  </div>

                  <div className='titleJoinForm'>
                    <Col sm={4} >
                      <label>Phone Number:</label>
                    </Col>
                    <Col sm={6}>
                      <Control.text
                        placeholder='Enter Your Phone Number'
                        className='input-group input-group-lg'
                        model='.phoneNumber'
                        validators={{
                          required,
                          isNumber,
                          maxLength: maxLength(15)
                        }}
                      />
                      <Errors
                        className='errors'
                        model='forms.join.phoneNumber'
                        show='touched'
                        messages={{
                          required: 'Required',
                          isNumber: ' Must be a phone number'
                        }}
                      />
                    </Col>
                  </div>
                  <div className='titleJoinForm'>
                    <Col sm={4} >
                      <label>Password:</label>
                    </Col>
                    <Col sm={6}>
                      <Control
                        placeholder='Enter Your Password'
                        className='input-group input-group-lg'
                        type='password'
                        model='.password'
                        validators={{
                          required,
                          maxLength: maxLength(15)
                        }}
                      />
                      <Errors
                        className='errors'
                        model='forms.join.password'
                        show='touched'
                        messages={{
                          required: 'Required',
                          maxLength: ' Must be 15 characters or less'
                        }}
                      />
                    </Col>
                  </div>

                  <div className='titleJoinForm'>
                    <Col sm={4} >
                      <label>Confirm Password:</label>
                    </Col>
                    <Col sm={6}>
                      <Control
                        placeholder='Re-Enter Your Password Here'
                        className='input-group input-group-lg'
                        type='password'
                        model='.confirmPassword'
                        validators={{
                          required,
                          maxLength: maxLength(15)
                        }}
                      />

                      <Errors
                        className='errors'
                        model='forms.join.confirmPassword'
                        show='touched'
                        messages={{
                          required: 'Required',
                          maxLength: ' Must be 15 characters or less'
                        }}
                      />
                    </Col>
                  </div>

                  <Errors
                    model='forms.join'
                    messages={{
                      passwordMatch: 'Password need to match'
                    }}
                    />
                  <div>
                    <Col sm={12} >
                      <button className='submitJoinBtn btn btn-default dropdown-toggle' type='submit'>Add</button>
                      <div id='successful-signup-confirm' />
                    </Col>
                  </div>
                </div>
              </Form>
            </div>
          </div>
        </div>
      </div>
    )
  }
})

Join.propTypes = {
  signUp: PropTypes.func.isRequired
}

export default Join
