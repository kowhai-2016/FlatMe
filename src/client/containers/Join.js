import { connect } from 'react-redux'

import { signUp } from '../actions'

import Join from '../components/Join'

const mapDispatchToProps = dispatch => {
  return {
    signUp: values => { return dispatch(signUp(values)) }
  }
}

export default connect(null, mapDispatchToProps)(Join)
