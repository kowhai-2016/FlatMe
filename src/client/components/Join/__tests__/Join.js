/* eslint-env jest */

import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import React from 'react'

import Join from '..'

test('Join renders correctly', () => {
  const wrapper = shallow(<Join />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})
