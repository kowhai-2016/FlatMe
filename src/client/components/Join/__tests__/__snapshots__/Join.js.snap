exports[`test Join renders correctly 1`] = `
<div>
  <h1>
    Join
  </h1>
</div>
`;
